### PR TITLE
avoid always compacting level0 under heavy write pressure

### LINKED ISSUE #6361: Minor compaction may just happens in level 0 under heavy write pressure
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)
set `profiles.default.dt_flush_after_write` to true, and check the log to find the minor compaction process.

### 2. What did you expect to see? (Required)
Minor compaction can happens in all level.

### 3. What did you see instead (Required)
Minor compaction happens mostly in level 0.

### PR DESCRIPTION
**Pull request by lidezhu**
### What problem does this PR solve?

Issue Number: close #6361 

Problem Summary: 
1. The minor compaction process may always select level 0 and there are two many small column files in other levels which cannot be compacted.
2. The minor compaction trigger condition is bad which will and it harder and harder to trigger it.

### What is changed and how it works?
1. Increase next compaction level before return compaction level;
2. Reset minor compaction trigger statistics if the compaction succeed;

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
Fix too many column files under heavy write pressure.
```

### CODE REVIEW

**Review by flowbehappy — APPROVED** [review]

**Review by Lloyd-Pottiger — APPROVED** [review]

**Review by JaySon-Huang — APPROVED** [review]
LGTM

File: dbms/src/Storages/DeltaMerge/Delta/ColumnFilePersistedSet.cpp [hunk 1]
```diff
@@ -324,6 +324,7 @@ MinorCompactionPtr ColumnFilePersistedSet::pickUpMinorCompaction(DMContext & con
 
         auto compaction = std::make_shared<MinorCompaction>(next_compaction_level, minor_compaction_version);
         auto & level = persisted_files_levels[next_compaction_level];
+        next_compaction_level++;
```
[the remark this answers is not in the document]
Comment (lidezhu) [reply dbms/src/Storages/DeltaMerge/Delta/ColumnFilePersistedSet.cpp:327]: I don't want to change it in cherry pick of release-6.1. So I prefer to just keep it here now.

**Pull request merged**